packages/aic/src/types.ts

```diff
@@ -0,0 +1,41 @@
+export type ErrorMessage = {
+    error: string
+    cause?: ErrorMessage
+    [key: string]: Value
+  }
+  
+export type Value =
+    | null
+    | boolean
+    | number
+    | string
+    | Value[]
+    | { [key: string]: Value }
+    | undefined
+
+export type TidString = string
+export type Signature = string
+export type DidKeyString = string
+export type Patch = ['put' | 'del', (string | number)[], Value]
```

Comment:
Similar to our MST encoding convo: do we want to avoid heterogenous arrays/tuples?

Should we prefer something like
```ts
type Patch = {
  op: 'put' | del'
  key: string[]
  val: Value
}
```